### PR TITLE
Update mimeType.php - load actual values. Update MimeTest for new data.

### PR DESCRIPTION
**Pull request by DeryabinSergey**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #19683

At this moment information in the source file https://svn.apache.org/viewvc/httpd/httpd/trunk/docs/conf/mime.types?view=markup changed like below
```
-$mimeTypes = [
+return [
     '3dml' => 'text/vnd.in3d.3dml',
     '3ds' => 'image/x-3ds',
     '3g2' => 'video/3gpp2',
@@ -318,11 +318,12 @@ $mimeTypes = [
     'htke' => 'application/vnd.kenameaapp',
     'htm' => 'text/html',
     'html' => 'text/html',
-    'hvd' => 'application/vnd.yamaha.hv-dic',
     'hvp' => 'application/vnd.yamaha.hv-voice',
     'hvs' => 'application/vnd.yamaha.hv-script',
     'i2g' => 'application/vnd.intergeo',
     'icc' => 'application/vnd.iccprofile',
+    0 => 'application/vnd.lotus-1-2-3',
+    'hvd' => 'application/vnd.yamaha.hv-dic',
     'ice' => 'x-conference/x-cooltalk',
     'icm' => 'application/vnd.iccprofile',
     'ico' => 'image/x-icon',
@@ -355,7 +356,6 @@ $mimeTypes = [
     'jam' => 'application/vnd.jam',
     'jar' => 'application/java-archive',
     'java' => 'text/x-java-source',
-    'jfif' => 'image/jpeg',
     'jisp' => 'application/vnd.jisp',
     'jlt' => 'application/vnd.hp-jlyt',
     'jnlp' => 'application/x-java-jnlp-file',
@@ -366,7 +366,7 @@ $mimeTypes = [
     'jpgm' => 'video/jpm',
     'jpgv' => 'video/jpeg',
     'jpm' => 'video/jpm',
-    'js' => 'application/javascript',
+    'js' => 'text/javascript',
     'json' => 'application/json',
     'jsonml' => 'application/jsonml+json',
     'kar' => 'audio/midi',
@@ -401,7 +401,6 @@ $mimeTypes = [
     'log' => 'text/plain',
     'lostxml' => 'application/lost+xml',
     'lrf' => 'application/octet-stream',
-    0 => 'application/vnd.lotus-1-2-3',
     'lrm' => 'application/vnd.ms-lrm',
     'ltf' => 'application/vnd.frogans.ltf',
     'lvp' => 'audio/vnd.lucent.voice',
@@ -996,9 +995,3 @@ $mimeTypes = [
     'zirz' => 'application/vnd.zul',
     'zmm' => 'application/vnd.handheld-entertainment+xml',
 ];
```

~Information about `xz` extension I move to `mimeType.php`, otherwise, my generated file was not correct for those, who use PHP versions lower than 8.01.~

~I don\`t know why at this moment in generated `mimeType.php` was a variable assignment, not return~

All fixed, replace move to generated block

### CODE REVIEW

File: build/controllers/MimeTypeController.php [hunk 1]
```diff
@@ -116,8 +112,13 @@ private function generateMimeTypesFile($outFile, $content)
  * https://svn.apache.org/viewvc/httpd/httpd/trunk/docs/conf/mime.types?view=markup
  * This file has been placed in the public domain for unlimited redistribution.
  */
-return $array;
+\$mimeTypes = $array;
+
+if (PHP_VERSION_ID >= 80100) {
```
Comment (samdark) [inline build/controllers/MimeTypeController.php:117]: Why moving it here?

Comment (DeryabinSergey) [reply build/controllers/MimeTypeController.php:117]: @samdark At actual version this condition placed here https://github.com/yiisoft/yii2/blob/master/framework/helpers/mimeTypes.php#L1000 but i think it was added here manually. This condition in controller added without manual testing ) https://github.com/yiisoft/yii2/blob/master/build/controllers/MimeTypeController.php#L105 and not working. First parameter in `array_replace` must be array, but passing string.

**Review by samdark — APPROVED** [review]
👍 but please add a line for CHANGELOG. Thanks.

**Comment by DeryabinSergey** [comment]
@samdark  ✅ done

**Review by Arhell — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**